test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -115,7 +116,7 @@ TEST_P(LogicalDnsParamTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized_, ready());
-  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
+  EXPECT_CALL(*dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
```

Comment:
Should we also have some tests for configuration, to ensure that the values that get put into a real JSON config fragment make their way to the LogicalDnsCluster object etc? See https://github.com/lyft/envoy/blob/master/test/server/configuration_impl_test.cc#L199 for an example of this kind of test.